Sources/PubNub/Events/New/Subscription.swift

```diff
@@ -165,35 +169,31 @@ extension Subscription: SubscribeMessagesReceiver {
   
   func event(from payload: SubscribeMessagePayload) -> PubNubEvent? {
     let isNewerOrEqualToTimetoken = payload.publishTimetoken.timetoken >= timetoken ?? 0
-    let receivedFromCurrentEntity: Bool
+    let isMatchingEntity: Bool
     
     if subscriptionType == .channel {
-      receivedFromCurrentEntity = entity.name.matches(string: payload.channel)
+      isMatchingEntity = isMatchingEntityName(entity.name, string: payload.channel)
     } else if subscriptionType == .channelGroup {
-      receivedFromCurrentEntity = entity.name.matches(string: payload.subscription ?? payload.channel)
+      isMatchingEntity = isMatchingEntityName(entity.name, string: payload.subscription ?? payload.channel)
     } else {
-      receivedFromCurrentEntity = true
+      isMatchingEntity = true
     }
     
-    if receivedFromCurrentEntity && isNewerOrEqualToTimetoken {
+    if isMatchingEntity && isNewerOrEqualToTimetoken {
       let event = payload.asPubNubEvent()
       return options.filterCriteriaSatisfied(event: event) ? event : nil
     } else {
       return nil
     }
   }
-}
-
-// MARK: - Helper String extension
-
-fileprivate extension String {
-  func matches(string: String) -> Bool {
-    guard hasSuffix(".*") else {
-      return self == string
+  
+  fileprivate func isMatchingEntityName(_ entityName: String, string: String) -> Bool {
+    guard entityName.hasSuffix(".*") else {
+      return entityName == string
     }
-    let pattern = "^" + self + "$"
-    let predicate = NSPredicate(format: "SELF MATCHES %@", pattern)
-
-    return predicate.evaluate(with: string)
+    if let firstIndex = entityName.lastIndex(of: "."), let secondIndex = string.lastIndex(of: ".") {
+      return entityName.prefix(upTo: firstIndex) == string.prefix(upTo: secondIndex)
+    }
```

Comment:
Isn't the "original" `NSPredicate` basically matched the same what is done on the line `#192`?

Reply:
Yeah, it should. It compares two String up to the last occurrence of `.`, assuming both contain the `.` character. Let's suppose these scenarios:

1. The message came from `channel.item.x` and the underlying entity's subscription is `channel.item.*`:
    -  According to the line 195, I will compare `channel.item` with `channel.item` ✅ 
2. The message came from `a.b.c` and the underlying entity's subscription is `channel.item.*`
    - I will compare `a.b` with `channel.item` 🔴

Reply:
Ok, I finally got it. 
I was totally fine with `extension` for `String` - it looked better.

I probably would use `payload.subscription ?? payload.channel` to match both `channel` and `channelGroup` subscription type. If a subscription has been created for wildcard subscription, then the `subscription` field will hold the same name (if an event for one of the channels from it) and will be no need for pattern match (just plain `==`). The only thing which may require some attention is presence channels / channel group names.